### ISSUE TITLE
desync not detected (LAN game)

### ISSUE THREAD

**Issue by tothxa** [widelands/widelands#4917]
**Describe the bug**
The game does not detect when host and client desync.

**To Reproduce**
1. Start LAN game with incompatible versions on the host and the client
2. Make it desync
3. Continue playing with divergent host and client

**Expected behavior**
The players should be notified and a warning should be printed on stdout when the desync happens

**Version:**
 - OS: antix / Debian buster + backports
 - Widelands Version: da98078 flatpak (flathub-beta) (host) vs. ca3ef43 (release) native (client)

**Additional context**
[stdout logs, replays and syncstreams](https://github.com/widelands/widelands/files/6593925/undetected_desync.zip)

I'm not sure if this is important enough for the 1.0 milestone. (edit: I mean whether it is reproducible by others)

(Originally I detected this with versions I thought to be compatible, because I forgot that the data dir of one of them had the lua scripts updated to recent master without recompiling.)

**Comment by Noordfrees** [comment]
The problem is not specific to LAN – it also happens when starting such a game via the metaserver but all clients are on the same physical machine. At a quick look the problem seems to be that clients with a very good real-time shift don't request sync reports often:
https://github.com/widelands/widelands/blob/7aa6186f9e5410fc35280c81a394d03a10dab04a/src/network/gamehost.cc#L1919-L1922
Solution would be to force-request a syncreport every now and then if the last one is already a long time past.

I believe this bug should not happen if the two LAN/IP players are sitting in different cities. So not critical IMHO.

Edit: According to the logs, the host requests a syncreport only once, directly at the beginning of the game:
```
[00:00:33.086 real] INFO: [Host]: Requesting sync reports for time 1
```

**Comment by tothxa** [comment]
While I created the example on a single machine, I observed it first in a real network game between two laptops next to each other, using wifi. But that's still low latency, of course.

**Comment by Noordfrees** [comment]
With the "recent" refactorings introducing lots of potential new bugs I really think this should be fixed for v1.1.

**Comment by matthiakl** [comment]
I noticed that the `CmdNetCheckSync` enqueued during the startup sync is never executed. Then the `syncreport_pending` is never reset, so no subsequent sync reports are done.

https://github.com/widelands/widelands/blob/aa65c4105cedf0c3b1545e0b6f27b694516d1d33/src/network/gamehost.cc#L2009-L2010

Any ideas how to debug it or why this command is not executed?
